src/services/point.service.ts

```diff
@@ -1,22 +1,61 @@
-import { Guild } from 'discord.js';
+import { Guild, Role, TextChannel, MessageEmbed, EmojiIdentifierResolvable } from 'discord.js';
 import { IUserPoints, Maybe } from '../common/types';
 import { PointsDocument, PointsModel } from '../schemas/points.schema';
 import { GuildService } from './guild.service';
+import Constants from '../common/constants';
+import { UserService } from './user.service';
 
 export class PointService {
   private _guild: Guild;
+  private _userservice: UserService;
+  private _tacoKingRole: Role;
+  private _tacoKingEmoji: Maybe<EmojiIdentifierResolvable> = null;
 
-  constructor(private _guildService: GuildService) {
+  constructor(private _guildService: GuildService, private _userService: UserService) {
     this._guild = this._guildService.get();
+    this._userservice = this._userService;
+    this._tacoKingRole = this._guild.roles.cache.find((r) => r.name === Constants.Roles.TacoKing)!;
+    this._tacoKingEmoji = this._guildService.getEmoji('tacoking') ?? '🌮';
   }
 
   public async awardPoints(id: Maybe<string>, amount: number) {
     if (!id) {
       return;
     }
 
+    // cache the #1 point holder before points are awarded
+    const kingBeforeAwardID = (await this.getTopPoints(1))[0].userID;
```

Comment:
```suggestion
    // cache the #1 point holder before points are awarded
    const [{ userID: kingBeforeAwardID }] = await this.getTopPoints(1);
```

This suggestion and one below are somewhat subjective. I find destructuring more elegant here, however it's up to you if want to accept this or not.